includes/reader-activation/sync/class-metadata.php

```diff
@@ -330,10 +332,25 @@ public static function normalize_contact_data( $contact ) {
 		}
 
 		$metadata            = $contact['metadata'];
+		$user                = ! empty( $metadata['account'] ) ? \get_user_by( 'id', $metadata['account'] ) : false;
 		$normalized_metadata = [];
 		$raw_keys            = self::get_raw_keys();
 		$prefixed_keys       = self::get_prefixed_keys();
 
+		// Registration-related fields.
+		if ( $user ) {
+			$registration_method             = ! empty( $metadata['registration_method'] ) ? $metadata['registration_method'] : \get_user_meta( $user->ID, Reader_Activation::REGISTRATION_METHOD, true );
+			$connected_account               = ! empty( $metadata['connected_account'] ) ? $metadata['connected_account'] : \get_user_meta( $user->ID, Reader_Activation::CONNECTED_ACCOUNT, true );
+			if ( ! empty( $registration_method ) ) {
+				$metadata['registration_method'] = $registration_method;
+			}
+			if ( $connected_account && in_array( $connected_account, Reader_Activation::SSO_REGISTRATION_METHODS ) ) {
+				$metadata['connected_account'] = $connected_account;
+			} elseif ( $registration_method && in_array( $registration_method, Reader_Activation::SSO_REGISTRATION_METHODS ) ) {
+				$metadata['connected_account'] = $registration_method;
+			}
+		}
```

Comment:
Given this method has so many different steps, I think we'd have better readability by running this in a separate method.

Reply:
We should probably do the same with the UTM capturing bit.

Reply:
7037b6c4